dist_q_learning/q_estimators.py

```diff
@@ -584,232 +596,177 @@ def update(self, history_batch):
+                q_targets = IV_is / h + future_qs * (h - 1) / h
+            if debug:
+                print("Doing update using IRE uncertainty...")
+                print(f"s=\n{states}")
+                print(f"action=\n{update_action}")
+                print(f"IRE q_targets combined=\n{q_targets}")
+            # TODO lr must be scalar... Also what?
+            # TODO - do this update at the end, rather than use the model
+            #  for the next transition estimate?
+            ire_lr = self.get_lr(ns=ire_ns)
+            if debug:
+                print(f"IRE LR=\n{ire_lr}")
+            self.update_estimator(
+                states=states,
+                action=update_action,
+                q_targets=q_targets,
+                horizon=None if self.horizon_type == "inf" else h,
+                lr=ire_lr)
+
+            # TEMP - skip 2nd update
+            continue
+
+            # Do the transition uncertainty estimate
+            # For scaling:
+            if not self.scaled and self.horizon_type == "finite":
+                max_q = geometric_sum(1., self.gamma, h)
+            elif not self.scaled:
+                max_q = geometric_sum(1., self.gamma, "inf")
+            else:
+                max_q = 1.
+
+            trans_ns, q_alphas, q_betas = self.model(
+                action=update_action, horizon=h).uncertainty_estimate(
+                states,
+                x_batch=conv_states,
+                y_batch=self.estimate(
+                    states=conv_states,
+                    action=update_action,
+                    target=True),
+                max_est_scaling=max_q,
+                debug=debug,
+            )
+
+            q_target_transitions = scipy.stats.beta.ppf(
+                self.quantile, q_alphas, q_betas)
+            if debug:
+                print(f"Trans Q quantile vals\n{q_target_transitions}")
+                print(f"{q_target_transitions.shape}, {states.shape}")
+            assert q_target_transitions.shape[0] == states.shape[0], (
+                f"{q_target_transitions.shape}, {states.shape}")
+            if max_q != 1.:
                 # TODO - right operation here?
-                if states.shape[0] <= 10:
-                    print(f"Transition Q values {q_target_transitions}")
                 q_target_transitions /= max_q
-                if states.shape[0] <= 10:
-                    print(f"Scaled: {q_target_transitions}")
-
-                # TODO - batch the learning rate!
-                if states.shape[0] <= 10:
+                if debug:
+                    print(f"Scaled:\n{q_target_transitions}")
                     print("Learning scaled transition Qs")
-                self.update_estimator(
-                    states[idxs], actions[idxs], q_target_transitions,
-                    lr=self.get_lr(ns=trans_ns),
-                    horizon=None if self.horizon_type == "inf" else h)
+            # TODO - batch the learning rate?
```

Comment:
This looks right to me on first glance. I am a bit unsure about the which learning rate, but what it is currently seems sensible